pkg/kubelet/kuberuntime/labels.go

```diff
@@ -38,6 +39,8 @@ const (
 	containerTerminationMessagePolicyLabel = "io.kubernetes.container.terminationMessagePolicy"
 	containerPreStopHandlerLabel           = "io.kubernetes.container.preStopHandler"
 	containerPortsLabel                    = "io.kubernetes.container.ports"
+
+	windowsHostProcessContainer = "microsoft.com/hostprocess-container"
```

Comment:
a vendor label domain here is surprising... is this the source of this label or is this driving existing behavior elsewhere?

Reply:
the comments indicate this is only expected while the feature is in alpha? what's the plan to transition this from an annotation to a field?

Reply:
> a vendor label domain here is surprising... is this the source of this label or is this driving existing behavior elsewhere?

The label is used by hcsshim: https://github.com/microsoft/hcsshim/blob/407147a8825c21b2f1f1ef4e53b24b27fb8632fb/internal/oci/uvm.go#L94  

@dcantah might be able to shed more light on the source

> the comments indicate this is only expected while the feature is in alpha? what's the plan to transition this from an annotation to a field?

Yes the plan is to integrate this directly into containerd instead of using the annotation.

Reply:
I'm not really sure if this is the right place for the annotation/to be setting it. For some background, the plan was if everything went smoothly and we were able to get a mirroring field on the OCI runtime spec for this, to translate the field added in this PR to the new OCI spec field. If the OCI changes fall through, ideally we'd check if the HostProcess field is set on the cri spec and set this annotation in the cri plugin/containerd. 

I think we can probably remove this

Reply:
Although for Alpha just in case there's any challenges/pushback with adding a change to set these annotations in the cri plugin (I can't see how but just to be safe) I think it's alright to keep this, but obviously I'm not the last say here.

Reply:
> The label is used by hcsshim: https://github.com/microsoft/hcsshim/blob/407147a8825c21b2f1f1ef4e53b24b27fb8632fb/internal/oci/uvm.go#L94

Is the kubelet wired directly to hcsshim on windows? if not, setting a label specifically for that runtime is weirdly coupling

> > what's the plan to transition this from an annotation to a field?
> 
> Yes the plan is to integrate this directly into containerd instead of using the annotation.

I more meant what protects against this code persisting beyond the alpha phase?

Reply:
> Is the kubelet wired directly to hcsshim on windows? 

Not for containerd, dockershim uses hcsshim directly in some cases but not for starting containers.

> I more meant what protects against this code persisting beyond the alpha phase?

The plan is to remove it when moving to beta.  I don't know exactly the best way to protect against persistence.  Would making it a requirement to remove it for graduation to Beta work?

Otherwise I agree this is a strange coupling and we could remove the annotations here.  This will require updates to containerd to parse the CRI fields before it could be used (where as with the annotations it is usable with no changes to containerd). Containerd will read from the CRI field and move the value to  the OCI annotations while we are waiting for https://github.com/opencontainers/runtime-spec/pull/1098 to be accepted in the OCI spec.

Reply:
> Would making it a requirement to remove it for graduation to Beta work?

yeah, that seems reasonable

Reply:
I've updated the KEP to reflect this: https://github.com/kubernetes/enhancements/pull/2755